### ISSUE TITLE
superlu-dist picks up system lapack instead of intel-mkl

### ISSUE THREAD

**Issue by s-sajid-ali** [spack/spack#10173]
`superlu-dist@6.1.0 ^intel-mkl %gcc` picks up system lapack instead of intel-mkl

@balay @gchavez2 @davydden 

```
$ spack install superlu_dist ^intel-mkl@2019.0.117 %gcc@7.3.0
```

### Platform and user environment

Please report your OS here:
```
$ uname -a 
Linux xrm 3.10.0-693.5.2.el7.x86_64 #1 SMP Fri Oct 13 10:46:25 EDT 2017 x86_64 x86_64 x86_64 GNU/Linux
$ lsb_release -d
Description:    Red Hat Enterprise Linux Workstation release 7.4 (Maipo)
``` 

### Build.out

[spack-build.out](https://pastebin.com/9FcZRBjr)

The issue is specifically in these lines : 
```
-- Using TPL_BLAS_LIBRARIES='-L/raid/home/sajid/packages/spack/opt/spack/linux-rhel7-x86_64/gcc-7.3.0/intel-mkl-2019.0.117-wzqlcijwx7odz2x5chembudo5leqpfh2/compilers_and_libraries_2019.0.117/linux/mkl/lib/intel64 -L/lib64 -lmkl_intel_lp64 -lmkl_sequential -lmkl_core -lpthread -lm -ldl'
-- Looking for Fortran sgemm
-- Looking for Fortran sgemm - found
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE
-- Found BLAS: /usr/lib64/libblas.so
```
@balay  was able to build this properly using intel compiler and intel-mkl so this might be an issue with cmake not picking up the libraries properly.

**Comment by davydden** [comment]
According to this line 
https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/superlu-dist/package.py#L46 we explicitly set blas/lapack. Yo should report the issue upstream, I think there is nothing Spack can do.

**Comment by s-sajid-ali** [comment]
Lines 8 and 9 of the build log show this: 
```
==> lapack_libs.blas_libs:  /raid/home/sajid/packages/spack/opt/spack/linux-rhel7-x86_64/gcc-7.3.0/intel-mkl-2019.0.117-wzqlcijwx7odz2x5chembudo5leqpfh2/compilers_and_libraries_2019.0.117/linux/mkl/lib/intel64/libmkl_intel_lp64.so /raid/home/sajid/packages/spack/opt/spack/linux-rhel7-x86_64/gcc-7.3.0/intel-mkl-2019.0.117-wzqlcijwx7odz2x5chembudo5leqpfh2/compilers_and_libraries_2019.0.117/linux/mkl/lib/intel64/libmkl_sequential.so /raid/home/sajid/packages/spack/opt/spack/linux-rhel7-x86_64/gcc-7.3.0/intel-mkl-2019.0.117-wzqlcijwx7odz2x5chembudo5leqpfh2/compilers_and_libraries_2019.0.117/linux/mkl/lib/intel64/libmkl_core.so
==> lapack_libs.blas_libs:  /lib64/libpthread.so /lib64/libm.so /lib64/libdl.so
```
And on line 18:
```
 '-DTPL_BLAS_LIBRARIES=-L/raid/home/sajid/packages/spack/opt/spack/linux-rhel7-x86_64/gcc-7.3.0/intel-mkl-2019.0.117-wzqlcijwx7odz2x5chembudo5leqpfh2/compilers_and_libraries_2019.0.117/linux/mkl/lib/intel64 -L/lib64 -lmkl_intel_lp64 -lmkl_sequential -lmkl_core -lpthread -lm -ldl'
```
Is that because spack didn't pick up blas/lapack correctly before passing the argument to `DTPL_BLAS_LIBRARIES` ?

**Comment by davydden** [comment]
> Is that because spack didn't pick up blas/lapack correctly

As far as I can tell, Spack does everything right, see [Intel MKL Link Advisor](https://software.intel.com/en-us/articles/intel-mkl-link-line-advisor). 

`-L/lib64` is because of the system libs (`/lib64/libm.so`), which got translated to `-L/lib64 -lm` etc.

There is [filter_system_paths](https://spack.readthedocs.io/en/latest/spack.util.html?highlight=filter_system#spack.util.environment.filter_system_paths) which is used in some places inside environment. I am not sure if we shall use it in `LibraryList` before returning from `ld_flags`. But this may not change anything for this issue.

**Comment by balay** [comment]
Now fixed by #11187
